mp/src/game/client/momentum/ui/HUD/hud_mapfinished.cpp

```diff
@@ -245,6 +259,12 @@ bool CHudMapFinishedDialog::ClosePanel()
     if (!m_bCanClose && !m_bIsGhost)
         return false;
 
+    // If player gets a new pb, update player best
+    if (m_flOverallDiff < 0.00f)
+    {
+        m_bPbNeedUpdate = true;
```

Comment:
This seems pretty event-based. Can loading the PB happen when the `replay_save` or `run_upload` events happen? Players aren't going to be getting a new PB every time they close the panel...

Reply:
If the PB gets updated before the panel closes you start losing the comparison values since the current time is the new PB. The other way i thought about dealing with this would be to store the comparisons as soon as the panel renders and not update it until the panel is opened again, but i thought putting a check at `ClosePanel` would be better.

Let me know if you think there's a better way of doing it.

Reply:
Yeah I think ClosePanel makes sense in that regard, then